pinot-core/src/main/java/org/apache/pinot/core/segment/index/loader/columnminmaxvalue/ColumnMinMaxValueGenerator.java

```diff
@@ -91,33 +92,46 @@ private void addColumnMinMaxValueForColumn(String columnName) throws Exception {
     switch (dataType) {
       case INT:
         try (IntDictionary intDictionary = new IntDictionary(dictionaryBuffer, length)) {
-          SegmentColumnarIndexCreator.addColumnMinMaxValueInfo(_segmentProperties, columnName,
-              intDictionary.getStringValue(0), intDictionary.getStringValue(length - 1));
+          SegmentColumnarIndexCreator
```

Comment:
How about no-dictionary columns? We do go over all the values, so we should be able to compute that information even without dictionary?

Reply:
The `ColumnMinMaxValueGenerator` is on server side (one of the pre-processors) generating column min/max value on the fly during segment load. We might not want to pay the cost of scanning all the values during segment load, so I kept the existing behavior.
The segment creation part is handled in `SegmentColumnarIndexCreator` where min/max values are added for both dictionary-encoded and raw index.